### PR TITLE
chore: update nebula and use hooks

### PR DESCRIPTION
**Pull request by Caele**
Update to latest Nebula version and refactor a bit to use hooks.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -10,21 +12,40 @@ export default function supernova(env) {
...
-      willUnmount() {},
-      destroy() {},
+    component: () => {
+      //  const [focusNodes, setFocusNodes] = useState([]);
+      const [dataTree, setDataTree] = useState({});
+      //   const [storage, setStorage] = useState({});
+      const layout = useStaleLayout();
+      const model = useModel();
+      const element = useElement();
+      const rect = useRect();
```
Comment (haxxmaxx) [inline src/index.js]: bounding rect?

Comment (Caele) [reply src/index.js]: basically the size of the chart

File: package.json [hunk 2]
```diff
@@ -46,7 +46,7 @@
     ]
   },
   "peerDependencies": {
-    "@nebula.js/supernova": "0.1.0-alpha.25"
+    "@nebula.js/supernova": "0.1.0-alpha.28"
```
Comment (miralemd) [inline package.json]:
```suggestion
    "@nebula.js/supernova": ">=0.1.0-alpha.28"
```

File: src/index.js [hunk 3]
```diff
@@ -10,21 +12,40 @@ export default function supernova(env) {
...
+
+      const callRender = () => {
+        if (element && dataTree && layout) {
+          return renderTree({
```
Comment (miralemd) [inline src/index.js]: A function returned from `useEffect` is a 'cleanup' function that will be executed when the specified deps change, and when the component is unmounted/destroyed - you probably need to split this method for usePromise and useEffect

File: src/index.js [hunk 4]
```diff
@@ -10,21 +12,40 @@ export default function supernova(env) {
...
+            element,
+            dataTree,
+            layout,
+            Theme,
```
Comment (miralemd) [inline src/index.js]: `env.Theme` is considered deprecated - `useTheme` instead

File: src/index.js (hunk 3 above)
Comment (Caele) [reply src/index.js]: think I want to use usePromise for all calls to render(), as they are likely to be async. Not needed now thought, and got into some infinite loops with it calling the rect one all the time.

**Pull request merged**